### PR TITLE
Display message after setting repl

### PR DESCRIPTION
**Pull request by jjttjj**
Displays a message after setting the repl.

When manually setting a repl with `inf-clojure-set-repl`, there is no visual indication that the repl has successfully been changed. This has confused me a few times in the past so I added a message.

This was [briefly discussed on slack](https://clojurians.slack.com/archives/C099W16KZ/p1672414905867309)

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines][1]
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

[1]: https://github.com/clojure-emacs/inf-clojure/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: inf-clojure.el [hunk 1]
```diff
@@ -285,10 +285,11 @@ buffers after they are created with `rename-buffer'."
...
     (let ((repl-buffers (inf-clojure-repls)))
-     (if (> (length repl-buffers) 0)
-         (when-let ((repl-buffer (completing-read "Select default REPL: " repl-buffers nil t)))
-           (setq inf-clojure-buffer (get-buffer repl-buffer)))
-       (user-error "No buffers have an inf-clojure process")))))
+      (if (> (length repl-buffers) 0)
+          (when-let ((repl-buffer (completing-read "Select default REPL: " repl-buffers nil t)))
+            (setq inf-clojure-buffer (get-buffer repl-buffer)))
+        (user-error "No buffers have an inf-clojure process"))))
+  (message "inf-clojure repl set to %s" inf-clojure-buffer))
```
Comment (bbatsov) [inline inf-clojure.el:292]: Probably it'd be better to use something like "Current/active inf-clojure REPL set to ...".

**Comment by dpsutton** [comment]
thanks for the addition. Great usability addition. Once you update to @bbatsov 's suggestion for the message we'll merge.

File: inf-clojure.el (hunk 1 above)
Comment (jjttjj) [reply inf-clojure.el]: Thanks for the input, I made the change!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## master (unreleased)
 * [#202](https://github.com/clojure-emacs/inf-clojure/issues/202): Add ClojureCLR support.
 * [#204](https://github.com/clojure-emacs/inf-clojure/issues/204): Scroll repl buffer on insert commands
+* Display message after setting repl.
```
Comment (dpsutton) [inline CHANGELOG.md:6]: you should add a link here 
> * [#208](https://github.com/clojure-emacs/inf-clojure/pull/208) Display message after setting repl with `inf-clojure-set-repl`

File: inf-clojure.el [hunk 3]
```diff
@@ -285,10 +285,11 @@ buffers after they are created with `rename-buffer'."
...
     (let ((repl-buffers (inf-clojure-repls)))
-     (if (> (length repl-buffers) 0)
-         (when-let ((repl-buffer (completing-read "Select default REPL: " repl-buffers nil t)))
-           (setq inf-clojure-buffer (get-buffer repl-buffer)))
-       (user-error "No buffers have an inf-clojure process")))))
+      (if (> (length repl-buffers) 0)
+          (when-let ((repl-buffer (completing-read "Select default REPL: " repl-buffers nil t)))
+            (setq inf-clojure-buffer (get-buffer repl-buffer)))
+        (user-error "No buffers have an inf-clojure process"))))
+  (message "Current inf-clojure REPL set to %s" inf-clojure-buffer))
```
Comment (dpsutton) [inline inf-clojure.el:292]: Also, should this `message` move this to be inside of the `when-let` which sets `inf-clojure-buffer`?

Comment (jjttjj) [reply inf-clojure.el]: You're right, I missed that possible outcome.
I changed it to only display the message after a successful change. Not confident in my style choices here, I could also just repeat the call to `message` in both success branches separately, let me know if you have a preference. 
https://github.com/jjttjj/inf-clojure/blob/04b967f1e60adb79cd84d580268f7af00021eb71/inf-clojure.el#L284-L294

Comment (dpsutton) [reply inf-clojure.el]: I don't have any strong opinions. I lean towards having the `message` and the `setq` in the same body so they work in tandem. But it doesn't really matter.

File: inf-clojure.el [hunk 4]
```diff
@@ -281,17 +281,17 @@ ALWAYS-ASK).  Otherwise get a list of all active inf-clojure
...
-                  (inf-clojure-repl-p))
-             (setq inf-clojure-buffer (current-buffer))
-           (let ((repl-buffers (inf-clojure-repls)))
-             (if (> (length repl-buffers) 0)
-                 (when-let ((repl-buffer (completing-read "Select default REPL: " repl-buffers nil t)))
-                   (setq inf-clojure-buffer (get-buffer repl-buffer)))
-               (user-error "No buffers have an inf-clojure process"))))))
-    (when new-repl-buffer
-      (message "Current inf-clojure REPL set to %s" new-repl-buffer))))
+  (if (and (not always-ask)
```
Comment (bbatsov) [inline inf-clojure.el:284]: The nest ifs make it a bit hard to follow the code, so some comments at the beginning of each branch might be useful.

File: inf-clojure.el [hunk 5]
```diff
@@ -281,17 +281,17 @@ ALWAYS-ASK).  Otherwise get a list of all active inf-clojure
...
+           (inf-clojure-repl-p))
+      (progn
+        (setq inf-clojure-buffer (current-buffer))
+        (message "Current inf-clojure REPL set to %s" inf-clojure-buffer))
```
Comment (bbatsov) [inline inf-clojure.el:288]: To reduce this message duplication you can just return the new-repl buffer at the end and set it outside the `if`s with a single message there. You can even the the code looking for the new repl to be a separate private function, so this would read a bit better. I'd prefer such less imperative code structure.

**Comment by jjttjj** [comment]
After trying a few arrangements, I went with extracting a function to just handle the "select repl" prompt. I believe this could also be reused to refactor [this](https://github.com/clojure-emacs/inf-clojure/blob/e5ce3839835b9b561fca5810f43f413c96c197d9/inf-clojure.el#L1484-L1501) if you would like. 
(Sorry if this little change has required a disproportionate amount of attention)

**Comment by bbatsov** [comment]
> After trying a few arrangements, I went with extracting a function to just handle the "select repl" prompt. I believe [this](https://github.com/clojure-emacs/inf-clojure/blob/e5ce3839835b9b561fca5810f43f413c96c197d9/inf-clojure.el#L1484-L1501) could also be reused to refactor this if you would like.

Yeah, it'd be nice to remove the code duplication, so a follow-up PR will be appreciated. 

>  (Sorry if this little change has required a disproportionate amount of attention)

No worries! The code is better in the end and that is all that matters.
